src/dynload/dynload.js

```diff
@@ -0,0 +1,247 @@
+const memoize = (fn) => {
+    let cache = {};
+    return (...args) => {
+        let n = args[0];
+        if (n in cache) {
+            return cache[n];
+        } else {
+            let result = fn(n);
+            cache[n] = result;
+            return result;
+        }
+    };
+};
+
+
+function createLock() {
+    let _lock = Promise.resolve();
+
+    async function acquireLock() {
+        const old_lock = _lock;
+        let releaseLock = () => { };
+        _lock = new Promise((resolve) => (releaseLock = resolve));
+        await old_lock;
+        return releaseLock;
+    }
+    return acquireLock;
+}
+
+function isInSharedLibraryPath(prefix, libPath){
+    if (libPath.startsWith("/")){
+        const dirname = libPath.substring(0, libPath.lastIndexOf("/"));
+        if(prefix == "/"){
+            return (dirname == `/lib`);
+        }
+        else{
+          return (dirname == `${prefix}/lib`);
+        }
+    }
+    else{
+        return false;
+    }
+}
+
+export async function loadDynlibsFromPackage(
+    prefix,
+    pkg_file_name,
+    pkg_is_shared_library,
+    dynlibPaths,
+    Module
+  ) {
+console.log('dynlibPaths',dynlibPaths);
```

Comment:
Let's clean up the `console.log`s that are not wrapped around a `verbose` variable check

Reply:
done